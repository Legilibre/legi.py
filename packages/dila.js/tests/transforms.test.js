const Legi = require("../src");
const markdown = require("../src/markdown");
const html = require("../src/html");
const knexConfig = require("../src/knexfile");

const legi = new Legi(knexConfig.test);

afterAll(() => {
  legi.close();
});

// medailles LEGITEXT000006070666
// travail LEGITEXT000006072050
// PI LEGITEXT000006069414

const CODE_TEST = "LEGITEXT000006070666";
//jest.setTimeout(20000);

it("getCode then markdown", async () => {
  expect.assertions(1);
  const res = await legi.getCode({ cid: CODE_TEST, date: "2018-12-01" }).then(markdown);
  expect(res).toMatchSnapshot();
});

it("getCode then html", async () => {
  expect.assertions(1);
  const res = await legi.getCode({ cid: CODE_TEST, date: "2018-12-01" }).then(html);
  expect(res).toMatchSnapshot();
});
