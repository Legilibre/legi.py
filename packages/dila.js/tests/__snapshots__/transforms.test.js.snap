// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCode then html 1`] = `
"<!DOCTYPE html>
<html lang=\\"fr\\">
<head>
<meta charset=\\"utf-8\\">
<title>legi.js</title>
<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
<link rel=\\"stylesheet\\" href=\\"https://rawgit.com/sindresorhus/github-markdown-css/gh-pages/github-markdown.css\\">
</head>
<body>
<div class=\\"markdown-body\\"><p><h1>Chapitre I : Monnaies et médailles</h1><br>
<br>
<p><h2>Section I : Fabrication des monnaies et médailles</h2><br>
<br>
<p><h3>Paragraphe II : Frappe des médailles.</h3><br>
<br>
<p><h4></h4><br>
<br>
<p>Il est expressément défendu à toutes personnes, quelles que soient les professions qu'elles exercent, de frapper ou de faire frapper des médailles, jetons ou pièces de plaisir, d'or, d'argent et autres métaux, ailleurs que dans les ateliers de la monnaie, à moins d'être munies d'une autorisation spéciale du ministre de l'économie et des finances.<br>
<br>
Néanmoins, tout dessinateur ou graveur ou autre personne peut dessiner ou graver, faire dessiner ou graver des médailles ; celles-ci sont frappées avec le coin qu'ils remettent à la Monnaie de Paris.<br>
<br>
Les frais de fabrication sont réglés par le ministre de l'économie et des finances.</p><br>
</p><p><h4></h4><br>
<br>
<p>Les particuliers qui font frapper des médailles ou jetons sont assujettis aux lois et règlements généraux de police qui concernent les arts et l'imprimerie.</p><br>
</p><p><h4></h4><br>
<br>
<p>Chacun des contrevenants aux dispositions de l'article 9 sera condamné à une amende de 3750 euros.</p><br>
</p><p><h4></h4><br>
<br>
<p>Les coins et poinçons de médailles déposés à l'hôtel des monnaies depuis plus de trente ans et dont les propriétaires ou ayants droit actuels sont inconnus peuvent être utilisés par cet établissement, à moins d'opposition des intéressés dans un délai de six mois à partir de la publication au Journal officiel d'un avis faisant connaître le sujet de la médaille, son module, le nom de l'artiste dont elle porte la signature, et l'année du dépôt.</p><br>
</p></p><p><h3>Paragraphe III : Dépôt légal.</h3><br>
<br>
<p><h4></h4><br>
<br>
<p>Il doit être déposé, tant à la Bibliothèque nationale qu'au musée monétaire de la Monnaie de Paris, deux exemplaires de chaque type nouveau de monnaie nationale et un exemplaire en bronze de chaque médaille nouvelle frappée à la Monnaie.<br>
<br>
Sous peine d'une amende prévue par le 5° de l'article 131-13 du code pénal pour les contraventions de la 5 ème classe pour chaque infraction dûment constatée, toute personne autorisée à frapper des médailles en dehors de la Monnaie devra déposer à la Bibliothèque nationale, d'une part, et au musée monétaire, d'autre part, dans le délai de quarante jours après la première frappe, un exemplaire de chaque médaille nouvelle choisi parmi les meilleurs au point de vue de la perfection d'exécution.</p><br>
</p></p></p></p><p><h1>Chapitre IV : Dispositions communes.</h1><br>
<br>
<p><h2></h2><br>
<br>
<p>Le présent code se substitue, dans les conditions prévues par la loi n° 51-1087 du 12 septembre 1951, aux dispositions des lois, ordonnances et décrets qui suivent :<br>
<br>
Décret des 3 et 20 septembre 1792.<br>
<br>
Décret des 8 et 9 novembre 1792, article 22.<br>
<br>
Décret du 25 thermidor An III.<br>
<br>
Arrêté du 5 germinal An XII.<br>
<br>
Décret du 11 mai 1807.<br>
<br>
Loi du 31 juillet 1879.<br>
<br>
Loi du 11 juillet 1885 (en ce qui concerne les billets de banque).<br>
<br>
Loi du 30 novembre 1896.<br>
<br>
Loi du 30 mars 1902, article 57.<br>
<br>
Loi du 29 mars 1904.<br>
<br>
Loi du 22 avril 1905, article 25.<br>
<br>
Loi du 30 janvier 1907, article 78.<br>
<br>
Loi du 31 décembre 1921, article 91.<br>
<br>
Loi du 28 juillet 1929, article unique.<br>
<br>
Loi du 8 avril 1931.<br>
<br>
Loi du 31 décembre 1937, article 53.<br>
<br>
Ordonnance du 26 août 1943, article 3.<br>
<br>
Ordonnance n° 45-164 du 2 février 1945.<br>
<br>
Loi n° 49-590 du 26 avril 1949.<br>
<br>
Loi n° 49-981 du 22 juillet 1949, article 35.<br>
<br>
Loi n° 50-586 du 27 mai 1950, articles 23,24,26,28.<br>
<br>
La loi n° 58-346 du 3 avril 1958 a attribué valeur législative au code des instruments monétaires et des médailles.</p><br>
</p></p></div>
</body>
</html>
"
`;

exports[`getCode then markdown 1`] = `
"# Chapitre I : Monnaies et médailles  
  
## Section I : Fabrication des monnaies et médailles  
  
### Paragraphe II : Frappe des médailles.  
  
####   
  
Il est expressément défendu à toutes personnes, quelles que soient les professions qu'elles exercent, de frapper ou de faire frapper des médailles, jetons ou pièces de plaisir, d'or, d'argent et autres métaux, ailleurs que dans les ateliers de la monnaie, à moins d'être munies d'une autorisation spéciale du ministre de l'économie et des finances.  
  
Néanmoins, tout dessinateur ou graveur ou autre personne peut dessiner ou graver, faire dessiner ou graver des médailles ; celles-ci sont frappées avec le coin qu'ils remettent à la Monnaie de Paris.  
  
Les frais de fabrication sont réglés par le ministre de l'économie et des finances.
  
####   
  
Les particuliers qui font frapper des médailles ou jetons sont assujettis aux lois et règlements généraux de police qui concernent les arts et l'imprimerie.
  
####   
  
Chacun des contrevenants aux dispositions de l'article 9 sera condamné à une amende de 3750 euros.
  
####   
  
Les coins et poinçons de médailles déposés à l'hôtel des monnaies depuis plus de trente ans et dont les propriétaires ou ayants droit actuels sont inconnus peuvent être utilisés par cet établissement, à moins d'opposition des intéressés dans un délai de six mois à partir de la publication au Journal officiel d'un avis faisant connaître le sujet de la médaille, son module, le nom de l'artiste dont elle porte la signature, et l'année du dépôt.
  
### Paragraphe III : Dépôt légal.  
  
####   
  
Il doit être déposé, tant à la Bibliothèque nationale qu'au musée monétaire de la Monnaie de Paris, deux exemplaires de chaque type nouveau de monnaie nationale et un exemplaire en bronze de chaque médaille nouvelle frappée à la Monnaie.  
  
Sous peine d'une amende prévue par le 5° de l'article 131-13 du code pénal pour les contraventions de la 5 ème classe pour chaque infraction dûment constatée, toute personne autorisée à frapper des médailles en dehors de la Monnaie devra déposer à la Bibliothèque nationale, d'une part, et au musée monétaire, d'autre part, dans le délai de quarante jours après la première frappe, un exemplaire de chaque médaille nouvelle choisi parmi les meilleurs au point de vue de la perfection d'exécution.
  
# Chapitre IV : Dispositions communes.  
  
##   
  
Le présent code se substitue, dans les conditions prévues par la loi n° 51-1087 du 12 septembre 1951, aux dispositions des lois, ordonnances et décrets qui suivent :  
  
Décret des 3 et 20 septembre 1792.  
  
Décret des 8 et 9 novembre 1792, article 22.  
  
Décret du 25 thermidor An III.  
  
Arrêté du 5 germinal An XII.  
  
Décret du 11 mai 1807.  
  
Loi du 31 juillet 1879.  
  
Loi du 11 juillet 1885 (en ce qui concerne les billets de banque).  
  
Loi du 30 novembre 1896.  
  
Loi du 30 mars 1902, article 57.  
  
Loi du 29 mars 1904.  
  
Loi du 22 avril 1905, article 25.  
  
Loi du 30 janvier 1907, article 78.  
  
Loi du 31 décembre 1921, article 91.  
  
Loi du 28 juillet 1929, article unique.  
  
Loi du 8 avril 1931.  
  
Loi du 31 décembre 1937, article 53.  
  
Ordonnance du 26 août 1943, article 3.  
  
Ordonnance n° 45-164 du 2 février 1945.  
  
Loi n° 49-590 du 26 avril 1949.  
  
Loi n° 49-981 du 22 juillet 1949, article 35.  
  
Loi n° 50-586 du 27 mai 1950, articles 23,24,26,28.  
  
La loi n° 58-346 du 3 avril 1958 a attribué valeur législative au code des instruments monétaires et des médailles.
  
"
`;
